### ISSUE TITLE
Railway tracks at "highway only" zoom levels

### ISSUE THREAD

**Issue by flxbt** [organicmaps/organicmaps#4945]
Hello!

As a frequent user of 'Organic Maps', I regularly miss an overview of railway connections: It would be great to have railway tracks added to the currently "highway only" zoom levels!

Thanks for your great app and best regards,
flxbt

**Comment by biodranik** [comment]
@flxbt what is your main use-case when you're overviewing railways? Do you plan your train trip in this way?

**Comment by flxbt** [comment]
I'd surely do, for example when searching to avoid the high-speed
lines, with tunnels and noisebarrier-shielded bridges alternating.

But I must confess that the reason for this feature request is also of
aesthetical or philosophical nature: I think that an "organic map"
deserves it to present an organic way of traveling on the "front page".

**Comment by pastk** [comment]
Duplicate of #2844

**Comment by pastk** [comment]
Railways will be more visible after #5510

Let's see if we can include main lines into the world overview map too..

**Comment by pastk** [comment]
> Let's see if we can include main lines into the world overview map too..

It turned out to be more involved as railways should be pre-processed first to look good at World overview levels https://github.com/organicmaps/organicmaps/pull/5548

**Comment by biodranik** [comment]
Adding railways to the world map would be great (maybe in a specific style, e.g. Public Transport one):

1|2
-|-
![Image](https://github.com/user-attachments/assets/f9266b03-868d-45e3-b888-562f4f2d2380)|![Image](https://github.com/user-attachments/assets/aa1dd856-fdfb-47fb-bc52-5212c8b7eece)

But the problem is how to display them in a usable/recognizable way:

![Image](https://github.com/user-attachments/assets/eacb29c8-13c2-499a-ba1d-6c2b13bc4727)

**Comment by zloidemon** [comment]
> Adding railways to the world map would be great (maybe in a specific style, e.g. Public Transport one):

[Alexander Borsuk](https://github.com/biodranik) Could you please clarify why railways should be classified as public transport, given that they serve more than just that purpose?

**Comment by flxbt** [comment]
An intuitively recognizable and (I think quite common) way of depicting railway lines is the black-and-white dashed line. 

-- 

[message sent otg]

**Comment by flxbt** [comment]
ooops, attachment not displayed... so here another example ("Eisenbahn" in line 4 w/o table header):

https://m.unser-stadtplan.de/stv-res/Legende_Stadtplan_DE.png

On 21 June 2025 15:05:09 UTC, Alexander Borsuk ***@***.***> wrote:
> biodranik left a comment (organicmaps/organicmaps#4945)
> 
> Adding railways to the world map would be great (maybe in a specific style, e.g. Public Transport one):
> 
> 1|2
> -|-
> ![Image](https://github.com/user-attachments/assets/f9266b03-868d-45e3-b888-562f4f2d2380)|![Image](https://github.com/user-attachments/assets/aa1dd856-fdfb-47fb-bc52-5212c8b7eece)
> 
> But the problem is how to display them in a usable/recognizable way:
> 
> ![Image](https://github.com/user-attachments/assets/eacb29c8-13c2-499a-ba1d-6c2b13bc4727)
> 


-- 

[message sent otg]

**Comment by biodranik** [comment]
@flxbt as we're speaking about world map zoom levels, multi-color approaches may not look great. It's better to test on real data/screenshot, as above.

**Comment by zloidemon** [comment]
> [@flxbt](https://github.com/flxbt) as we're speaking about world map zoom levels, multi-color approaches may not look great. It's better to test on real data/screenshot, as above.

[Alexander Borsuk](https://github.com/biodranik) Could we review if other applications or maps have implemented similar features and consider adopting the best approach?

**Comment by flxbt** [comment]
Please be aware, that my original title (and intension) of this "issue" has been changed by 'pastk' on July 19, 2023 and is now overshooting the (my) mark by far.



On 22 June 2025 10:56:26 UTC, Alexander Borsuk ***@***.***> wrote:
> biodranik left a comment (organicmaps/organicmaps#4945)
> 
> @flxbt as we're speaking about world map zoom levels, multi-color approaches may not look great. It's better to test on real data/screenshot, as above.
> 


-- 

[message sent otg]

**Comment by biodranik** [comment]
@flxbt main country highways are rendered from zoom level 6, and some other significant highways are rendered from zoom 8. Detailed country map downloads/draws at zoom 10. What was the scope of your initial proposal? You can see zoom levels by typing `?debug-info` in OM's search box.

**Comment by flxbt** [comment]
@biodranik The `?debug-info` feature, with and without backticks,
doesn't work for me (latest F-Droid version). But apart from that, I'd
suggest to handle it like the highways: Show main routes (e.g.
high-speed and international connections) beginning with zoom level 6
and go into more detail (long distance -> regional -> local) with
increasing levels. (Disclaimer: I have not a slight idea how
complicated it is to extract this info from the OSM data...)

BTW: I wasn't aware that the lower "highway zoom levels" are part of
the world map. But as, for my opinion, railway lines are visible very
badly in the highest zoom levels, too, I'd suggest to use exactly this
style for the world map and a similar, but considerable bolder one for
the higher zoom levels.

**Comment by biodranik** [comment]
?debug-info is typed without backticks or spaces in the search box. It enables showing fps and zoom level in the top left corner. 

As it was already shown on a [screenshot above](https://github.com/organicmaps/organicmaps/issues/4945#issuecomment-2993624919), drawing railways starting from zoom 6 makes the map unreadable. The problem is how to keep the map usable and readable for most of the users, including drivers for example.

**Comment by flxbt** [comment]
@biodranik

Nevermind, I've left for the 'CoMaps' fork anyway.

https://www.comaps.app/ 
https://openletter.earth/open-letter-to-organic-maps-shareholders-a0bf770c 
https://lwn.net/Articles/1024387/ 

Still, I'd highly appreciate some info / links regarding your (team's)
point of view about this fork.
